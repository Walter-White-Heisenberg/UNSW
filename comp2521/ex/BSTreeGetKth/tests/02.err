==40190== Memcheck, a memory error detector
==40190== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==40190== Using Valgrind-3.13.0 and LibVEX; rerun with -h for copyright info
==40190== Command: ./testBSTreeGetKth
==40190== 
==40190== Invalid read of size 8
==40190==    at 0x1093AB: BSTreeGetKth (BSTreeGetKth.c:31)
==40190==    by 0x109256: main (testBSTreeGetKth.c:18)
==40190==  Address 0x8 is not stack'd, malloc'd or (recently) free'd
==40190== 
==40190== 
==40190== Process terminating with default action of signal 11 (SIGSEGV)
==40190==  Access not within mapped region at address 0x8
==40190==    at 0x1093AB: BSTreeGetKth (BSTreeGetKth.c:31)
==40190==    by 0x109256: main (testBSTreeGetKth.c:18)
==40190==  If you believe this happened as a result of a stack
==40190==  overflow in your program's main thread (unlikely but
==40190==  possible), you can try to increase the size of the
==40190==  main thread stack using the --main-stacksize= flag.
==40190==  The main thread stack size used in this run was 8388608.
==40190== 
==40190== HEAP SUMMARY:
==40190==     in use at exit: 168 bytes in 7 blocks
==40190==   total heap usage: 18 allocs, 11 frees, 16,933 bytes allocated
==40190== 
==40190== LEAK SUMMARY:
==40190==    definitely lost: 0 bytes in 0 blocks
==40190==    indirectly lost: 0 bytes in 0 blocks
==40190==      possibly lost: 0 bytes in 0 blocks
==40190==    still reachable: 168 bytes in 7 blocks
==40190==         suppressed: 0 bytes in 0 blocks
==40190== Rerun with --leak-check=full to see details of leaked memory
==40190== 
==40190== For counts of detected and suppressed errors, rerun with: -v
==40190== ERROR SUMMARY: 1 errors from 1 contexts (suppressed: 0 from 0)
