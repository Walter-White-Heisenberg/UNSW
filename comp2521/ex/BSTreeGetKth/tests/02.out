Enter the preorder traversal of the BST: Tree:

   8
  / \
 /   \
2    12
 \     \
  5    13
 / \
4   7

Enter k: For k = 0, BSTreeGetKth returned 2
Enter k: For k = 1, BSTreeGetKth returned 5
Enter k: 